### PR TITLE
Several common 'household' reagents can be used as improvised medicine treatment. Updated first aid analyzer information.

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Several common 'household' reagents can be used as improvised medicine treatment.

Drinking tea will help mend (non-bone) wounds over time.

Flour and corn starch may be splashed onto wounds to help dry them up, though they'll have a negative effect on burn wounds.

Added a new reagent, saltwater, made by combining table salt with water.

Table salt and saltwater can be splashed onto wounds as well, reducing bleeding and improving sanitization and disinfection significantly. However, the coarse undiluted salt will irritate the wounds, reducing clot rate and flesh healing, and both of the reagents will increase a burn wound's infestation rate.

Altered Table Salt's recipe to just need sodium and chloride. Changed the recipe of Pentetic Acid and Heparin to need table salt (sodium x chloride) and thus slightly altered the total output of those reagents (pentacid went from 5u per reaction to 4u, heparin 4u->3u)

Saline-Glucose Solution now needs 2u of saltwater and 1u of sugar, meaning the overall recipe should be completely unchanged in practice. Contact me on discord if any issues arise from these chemical changes!

First aid analyzers now give easy-to-understand direct information, with the specific recommended treatments bolded in the analysis text. They also have a 'unique' extra bit of info, telling you about improvised ways to remedy your wound.
## Why It's Good For The Game

Wounds have a very poor amount of interaction with the rest of the game and have not had much added to them post-merge, especially in 'improvised' ways to help Not Die to a wound while you crawl your way to a emergency medkit or medbay. I researched info on this and found some interesting ideas - some of them I'll have to leave for later because this PR kept growing out of scope (Improvised bone gel, ice on wounds which turned into wound temperature mechanics, crutches, a 'suture item' component refactor...)

As for what this actually does to benefit the game, it allows more dynamic wound Gameplay as people use first aid analyzers to get information on treatment when medbay blows up, helps them stabilize by splashing flour onto themselves before looking for some actual treatment, helps traitors realize how they can self-treat many crippling wounds (at risk, of course). It expands treatment options to things beside medkits and medbay, but always does so in ways that have downsides that make them not ideal as _treatment_, and more beneficial as stabilizing before seeking true professional help. This thus significantly increases the rather shallow depth of wounds as a system to interact with.

> Several common 'household' reagents can be used as improvised medicine treatment.

From what I could tell by looking at several sources for each 'realistic' treatment, these are indeed semi-reasonable things that are done to wounds by some people as household treatment.

It goes without saying that you should **not do any of these things in real life** without consulting a doctor unless your blood is also spilling out by the gallon into the floor. All these 'realistic treatments' have drastic downsides and are meant for the short-term only. Except the tea.

> Drinking tea will help mend (non-bone) wounds over time.

Tea is healthy, we all know that.

> Flour and corn starch may be splashed onto wounds to help dry them up, though they'll have a negative effect on burn wounds.

Flour and apparently starch dries wounds up but risks infection. That's not a thing for blood wounds yet but oh well.

> Table salt and saltwater can be splashed onto wounds as well, reducing bleeding and improving sanitization and disinfection significantly. However, the coarse undiluted salt will irritate the wounds, reducing clot rate and flesh healing, and both of the reagents will increase a burn wound's infestation rate.

Salt kills bacteria via osmosis, but it also kills your own cells, and some bacteria like salt.

> Added a new reagent, saltwater, made by combining table salt with water.

> Altered Table Salt's recipe to just need sodium and chloride. Changed the recipe of Pentetic Acid and Heparin to need table salt (sodium x chloride) and thus slightly altered the total output of those reagents (pentacid went from 5u per reaction to 4u, heparin 4u->3u)

> Saline-Glucose Solution now needs 2u of saltwater and 1u of sugar, meaning the overall recipe should be completely unchanged in practice. Contact me on discord if any issues arise from these chemical changes!

I wish I hadn't had to mess with reagents like this, but I needed to because just adding mixing salt and water caused the saline glucose recipe to basically split itself into half saltwater half glucose. 

I removed the water requirement for table salt (Why did it even have that, salt ain't wet bro?), made saline-glucose need 2u saltwater and 1u sugar, and altered relevant recipes so they didn't also cause unwanted table salt to react from their sodium and chloride ingredients.

A happy side-effect is that saline glucose solution is even easier to make now as an improvised chem by mixing salt, water, and sugar, which fits pretty perfectly (especially as a temporary blood substitute)

> First aid analyzers now give easy-to-understand direct information, with the specific recommended treatments bolded in the analysis text. They also have a 'unique' extra bit of info, telling you about improvised ways to remedy your wound.

You might notice that as the wounds get more serious the text gets more direct and concise and reluctantly hands out more and more improvised treatment options, so that's fun. As for the improvised section itself, it helps people be actually aware of these ways to help treat themselves rather than delegating it to hyper-gamer knowledge.

The bolded treatment bit is pretty neat and means your eyes can inmediately focus on what you can do to save yourself - very useful if you have a weeping avulsion and no bandages.
## Changelog
:cl:
add: Several common 'household' reagents can be used as improvised medicine treatment.
add: Drinking tea will help mend (non-bone) wounds over time.
add: Flour and corn starch may be splashed onto wounds to help dry them up, though they'll have a negative effect on burn wounds.
add: Added a new reagent, saltwater, made by combining table salt with water.
add: Table salt and saltwater can be splashed onto wounds as well, reducing bleeding and improving sanitization and disinfection significantly. However, the coarse undiluted salt will irritate the wounds, reducing clot rate and flesh healing, and both of the reagents will increase a burn wound's infestation rate.
add: Altered Table Salt's recipe to just need sodium and chloride. Changed the recipe of Pentetic Acid and Heparin to need table salt (sodium x chloride) and thus slightly altered the total output of those reagents (pentacid went from 5u per reaction to 4u, heparin 4u->3u)
add: Saline-Glucose Solution now needs 2u of saltwater and 1u of sugar, meaning the overall recipe should be completely unchanged in practice. Contact me on discord if any issues arise from these chemical changes!
qol: First aid analyzers now give easy-to-understand direct information, with the specific recommended treatments bolded in the analysis text. They also have a 'unique' extra bit of info, telling you about improvised ways to remedy your wound.
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
I'm pretty sure the wounds system was partly implemented to make you go _into_ medbay, but this seems to be trying to take you out of it
I'm not entirely opposed to things like the corn starch thing giving a slight boost, but the tea one is a bit much imo. Tea already cures toxin damage, physical wounds is a bit much for it.

Another question; how do these compare to actual medical remedies? Are these slower, and if so how much?

**Comment by carlarctg** [comment]
> I'm pretty sure the wounds system was partly implemented to make you go _into_ medbay, but this seems to be trying to take you out of it

My man! Everything here is meant to STABILIZE you. Not to fix your wounds instantly. I know the wounds system is designed so that people don't just self treat everything. This is why blood loss reduction is clamped for salt. This is why salt increases infestation rate by a quite large amount, which is deadly if left to fester.

These things help people Not Die, it doesn't help them fix every issue immediately.

> I'm not entirely opposed to things like the corn starch thing giving a slight boost, but the tea one is a bit much imo. Tea already cures toxin damage, physical wounds is a bit much for it.

It's healthy. The numbers are honestly pretty goofy but I got tired of stat tuning- I should probably go back and make it just slowly increase clot rate and decrease infestation rate.

> Another question; how do these compare to actual medical remedies? Are these slower, and if so how much?

Again pretty much missing the point - none of these treat the wound, they prevent death. Space cleaner is a third as effective as sterilizine, and nothing is more effective than scanning, bandaging, and ointment or suturing.

File: code/datums/wounds/_wounds.dm [hunk 1]
```diff
@@ -24,6 +24,14 @@
 	/// What the limb looks like on a cursory examine
 	var/examine_desc = "is badly hurt"
 
+	/// Simple description, shortened for clarity if defined. Otherwise just takes the normal desc in the analyzer proc.
```
Comment (carlarctg) [inline code/datums/wounds/_wounds.dm:27]: forgot to implement

**Comment by carlarctg** [comment]
> You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

Not really sure what to do here. Mantainers please give me ideas. Bug hunters pick the low hanging fruit and I'm not sure what might be fun to refactor. Something I have planned is reworking sutures and meshes into a component that can be applied onto any item (Bone gel for skellies!)

**Comment by Wallemations** [comment]
If you want more gbp just help turn some mobs to basic mobs or something, do some QOL stuff like swapping chat alerts to bubble ones, etc

**Comment by Wallemations** [comment]
> > I'm pretty sure the wounds system was partly implemented to make you go _into_ medbay, but this seems to be trying to take you out of it
> 
> 
> 
> My man! Everything here is meant to STABILIZE you. Not to fix your wounds instantly. I know the wounds system is designed so that people don't just self treat everything. This is why blood loss reduction is clamped for salt. This is why salt increases infestation rate by a quite large amount, which is deadly if left to fester.
> 
> 
> 
> These things help people Not Die, it doesn't help them fix every issue immediately.
> 
> 
> 
> > I'm not entirely opposed to things like the corn starch thing giving a slight boost, but the tea one is a bit much imo. Tea already cures toxin damage, physical wounds is a bit much for it.
> 
> 
> 
> It's healthy. The numbers are honestly pretty goofy but I got tired of stat tuning- I should probably go back and make it just slowly increase clot rate and decrease infestation rate.
> 
> 
> 
> > Another question; how do these compare to actual medical remedies? Are these slower, and if so how much?
> 
> 
> 
> Again pretty much missing the point - none of these treat the wound, they prevent death. Space cleaner is a third as effective as sterilizine, and nothing is more effective than scanning, bandaging, and ointment or suturing.

I admit I misinterpreted the pr, but again I stand by the tea part not making much sense beyond just "it's healthy". Yes it's healthy, which is why it fixes toxin damage, because that's what tea is said to do IRL. 
I'm not a code maint so I won't comment on any of that other stuff or any balance things, but you get my point about the tea.

**Comment by githubuser4141** [comment]
> 

'Stabilize' in /tg/'s medical system is a bit meh because lethal vs non-lethal is based on how much damage numbers you have. If it's 100 or above, you ded. If not, you fine. /Tg/'s medic system is bizarre where things that'd be trivial irl can easily kill you (e.g., scratching at an itch! x10), irl scratching nonstop is harmful but it's not instantly deadly - it COULD technically kill you via long-term causes I guess, and OTOH things that'd be very dangerous IRL can be easily stabilized and treated like gunshot wounds, slashing open someone's arteries, having an entire armoured vehicle falling on your head, etc. In /tg/, you're more likely to die from 'oops I mislicked and touched a hot lightbulb when bandgaging myself at 99 damage' then the actual wounds which caused 99 damage. 

Also source on the tea curing toxin damage irl? If you had a very small dosage of poison you might be able to get away with it by drinking large quantities of tea because it'd be diluted, but this goes for water too. If you're actually dying from poison induced damage then I'm sorry but tea isn't going to help

**Comment by Likteer** [comment]
obraz

**Comment by Dorsisdwarf** [comment]
Im not quite sure how saltwater works? it disinfects and reduces infection, but also speeds up infections on the only wound type that can be infected? How do all those interact? 

also saltwater recipe will break porridge and electron soup recipies, those both use water and salt as reagants

**Comment by BlueMemesauce** [comment]
Just change everything that uses salt and water to use saltwater instead

**Comment by carlarctg** [comment]
> Im not quite sure how saltwater works? it disinfects and reduces infection, but also speeds up infections on the only wound type that can be infected? How do all those interact?
> 
> also saltwater recipe will break porridge and electron soup recipies, those both use water and salt as reagants

Thanks, I'll change that up 

> Just change everything that uses salt and water to use saltwater instead

Yeah I did that with sodium and chloride

> obraz

uhhhh ok?

> > 
> 
> 'Stabilize' in /tg/'s medical system is a bit meh because lethal vs non-lethal is based on how much damage numbers you have. If it's 100 or above, you ded. If not, you fine. /Tg/'s medic system is bizarre where things that'd be trivial irl can easily kill you (e.g., scratching at an itch! x10), irl scratching nonstop is harmful but it's not instantly deadly - it COULD technically kill you via long-term causes I guess, and OTOH things that'd be very dangerous IRL can be easily stabilized and treated like gunshot wounds, slashing open someone's arteries, having an entire armoured vehicle falling on your head, etc. In /tg/, you're more likely to die from 'oops I mislicked and touched a hot lightbulb when bandgaging myself at 99 damage' then the actual wounds which caused 99 damage. I could stabilize my wounds with epi, salt water, flour, dirt, inhaling cattle medicine and maybe the odd orange juice IV but I'd still die from a random itch, or from a mouse biting your jackbooted toe.

What is this insane rant. I can't fix tg medical system in one small PR (Or at all). The 'stabilizing' in this case is just making sure your burn wounds don't get septic before you reach medbay or making sure you don't lose so much blood before the wound is fixed that you fall unconscious and die before you can get proper treatment and blood replenishment.

> Also source on the tea curing toxin damage irl? If you had a very small dosage of poison you might be able to get away with it by drinking large quantities of tea because it'd be diluted, but this goes for water too. If you're actually dying from poison induced damage then I'm sorry but tea isn't going to help

There is no source. Tea is healthy.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -281,6 +281,54 @@
 	//You don't belong in this world, monster!
 	mytray.reagents.remove_reagent(type, volume)
 
+/datum/reagent/water/salt
+	name = "Saltwater"
+	description = "Water, but salty. Smells like... Central command shuttle docking?"
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:286]:
```suggestion
	description = "Water, but salty. Smells like... the station infirmary?"
```

i think your current description is way too meta but this one has a nice double meaning

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -602,9 +638,42 @@
...
+	to_chat(carbies, span_notice("The flour seeps into [lowertext(src)], painfully drying it up and absorbing some of the blood."))
+	// When some nerd adds infection for wounds, make this increase the infection
+	return
+
+/datum/wound/slash/on_flour(reac_volume, mob/living/carbon/carbies)
+	adjust_blood_flow(-0.04 * reac_volume) // 30u of a flour sack * 0.04 = -1.25~ blood flow, pretty good!
+	to_chat(carbies, span_notice("The flour seeps into [lowertext(src)], painfully drying some of it up and absorbing a little blood."))
+	// When some nerd adds infection for wounds, make this increase the infection
+	return
+
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:667]:
```suggestion
	// When some nerd adds infection for wounds, make this increase the infection
```

no need for the redundant return

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 4]
```diff
@@ -602,9 +638,42 @@
...
+		iter_wound.on_flour(reac_volume, carbies)
+
+/datum/wound/proc/on_flour(reac_volume, mob/living/carbon/carbies)
+	return
+
+/datum/wound/pierce/on_flour(reac_volume, mob/living/carbon/carbies)
+	adjust_blood_flow(-0.015 * reac_volume) // 30u of a flour sack * 0.015 = -0.45~ blood flow, prettay good
+	to_chat(carbies, span_notice("The flour seeps into [lowertext(src)], painfully drying it up and absorbing some of the blood."))
+	// When some nerd adds infection for wounds, make this increase the infection
+	return
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:660]:
```suggestion
	// When some nerd adds infection for wounds, make this increase the infection
```

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 5]
```diff
@@ -442,6 +442,42 @@
...
+	to_chat(carbies, span_notice("The salt bits seep in and stick to [lowertext(src)], painfully irritating the skin but soaking up most of the blood."))
+
+/datum/wound/burn/on_salt(reac_volume)
+	// Slightly sanitizes and disinfects, but also increases infestation rate (some bacteria are aided by salt), and decreases flesh healing (can damage the skin from moisture absorption)
+	sanitization += VALUE_PER(0.4, 30) * reac_volume
+	infestation -= max(VALUE_PER(0.3, 30) * reac_volume, 0)
+	infestation_rate += VALUE_PER(0.12, 30) * reac_volume
+	flesh_healing -= max(VALUE_PER(5, 30) * reac_volume, 0)
+	to_chat(victim, span_notice("The salt bits seep in and stick to [lowertext(src)], painfully irritating the skin! After a few moments, it feels marginally better."))
+	return
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:478]:
```suggestion
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 6]
```diff
@@ -281,6 +281,54 @@
...
+	icon_state = "glass_clear"
+
+/datum/reagent/water/salt/expose_mob(mob/living/exposed_mob, methods, reac_volume)
+	. = ..()
+	var/mob/living/carbon/carbies = exposed_mob
+	if(!(methods & (PATCH|TOUCH|VAPOR)))
+		return
+	for(var/i in carbies.all_wounds)
+		var/datum/wound/iter_wound = i
+		iter_wound.on_saltwater(reac_volume, carbies)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:310]:
```suggestion
	for(var/datum/wound/iterable_wound as anything in carbies.all_wounds)
		iter_wound.on_saltwater(reac_volume, carbies)
```

this is faster and cleaner. if `all_wounds` carries things that aren't types of `/datum/wound` then remove the `as anything`.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 7]
```diff
@@ -669,6 +738,38 @@
...
+
+// Starch has similar absorbing properties to flour (Stronger here because it's rarer)
+/datum/reagent/consumable/corn_starch/expose_mob(mob/living/exposed_mob, methods, reac_volume)
+	. = ..()
+	var/mob/living/carbon/carbies = exposed_mob
+	if(!(methods & (PATCH|TOUCH|VAPOR)))
+		return
+	for(var/i in carbies.all_wounds)
+		var/datum/wound/iter_wound = i
+		iter_wound.on_starch(reac_volume, carbies)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:751]: ditto type + `as anything`

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 8]
```diff
@@ -602,9 +638,42 @@
...
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED|REAGENT_AFFECTS_WOUNDS
 	default_container = /obj/item/reagent_containers/condiment/flour
 
+/datum/reagent/consumable/flour/expose_mob(mob/living/exposed_mob, methods, reac_volume)
+	. = ..()
+	var/mob/living/carbon/carbies = exposed_mob
+	if(!(methods & (PATCH|TOUCH|VAPOR)))
+		return
+	for(var/i in carbies.all_wounds)
+		var/datum/wound/iter_wound = i
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:650]: ditto `type` and `as_anything`

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 9]
```diff
@@ -442,6 +442,42 @@
...
 	exposed_turf.spawn_unique_cleanable(/obj/effect/decal/cleanable/food/salt)
 
+/datum/reagent/consumable/salt/expose_mob(mob/living/exposed_mob, methods, reac_volume)
+	. = ..()
+	var/mob/living/carbon/carbies = exposed_mob
+	if(!(methods & (PATCH|TOUCH|VAPOR)))
+		return
+	for(var/i in carbies.all_wounds)
+		var/datum/wound/iter_wound = i
+		iter_wound.on_salt(reac_volume, carbies)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:452]: ditto type and `as anything`

File: code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm [hunk 10]
```diff
@@ -296,10 +296,40 @@
 	affected_mob.AdjustSleeping(-20 * REM * seconds_per_tick)
 	if(affected_mob.getToxLoss() && SPT_PROB(10, seconds_per_tick))
 		affected_mob.adjustToxLoss(-1, FALSE, required_biotype = affected_biotype)
+	var/to_chatted = FALSE
+	for(var/i in affected_mob.all_wounds)
+		var/datum/wound/iter_wound = i
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:301]: ditto type and `as anything`

File: code/game/objects/items/devices/scanners/health_analyzer.dm [hunk 11]
```diff
@@ -505,15 +505,15 @@
...
+			render_list += "<div class='ml-2'>[simple_scan ? current_wound.get_simple_scanner_description() : current_wound.get_scanner_description()]</div>\n"
 			ADD_TRAIT(current_wound, TRAIT_WOUND_SCANNED, ANALYZER_TRAIT)
 			if(!advised)
 				to_chat(user, span_notice("You notice how bright holo-images appear over your [(length(wounded_part.wounds) || length(patient.get_wounded_bodyparts()) ) > 1 ? "various wounds" : "wound"]. They seem to be filled with helpful information, this should make treatment easier!"))
 				advised = TRUE
 		render_list += "</span>"
 
 	if(render_list == "")
-		if(istype(scanner))
+		if(simple_scan == TRUE)
```
Comment (san7890) [inline code/game/objects/items/devices/scanners/health_analyzer.dm:516]:
```suggestion
		if(simple_scan)
```

no need to explicitly match a boolean value if we know it's a boolean

File: code/datums/wounds/burns.dm [hunk 12]
```diff
@@ -40,14 +40,12 @@
...
 
-	if(victim.reagents)
-		if(HAS_TRAIT(victim, TRAIT_VIRUS_RESISTANCE))
-			sanitization += 0.9
-		if(victim.reagents.has_reagent(/datum/reagent/space_cleaner/sterilizine/))
-			sanitization += 0.9
-		if(victim.reagents.has_reagent(/datum/reagent/medicine/mine_salve))
-			sanitization += 0.3
-			flesh_healing += 0.5
+	for(var/datum/reagent/reagent in victim.reagents.reagent_list)
```
Comment (san7890) [inline code/datums/wounds/burns.dm:43]: would `as anything` work here?

File: code/datums/wounds/_wounds.dm [hunk 13]
```diff
@@ -24,6 +24,14 @@
...
+	var/s_desc
+	/// Simple analyzer's wound description, which focuses less on the clinical aspect of the wound and more on easily readable treatment instructions.
+	var/s_treat_text = "Go to medbay idiot"
+	/// Improvised remedies indicated by the first aid analyzer only.
+	var/homemade_treat_text = "Remember to drink lots of water!"
```
Comment (san7890) [inline code/datums/wounds/_wounds.dm:32]: spell out what `s` might be, so `simple_desc` and `simple_treat_text`. letters are free.

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by carlarctg** [comment]
@san7890 Aside from the code nitpicks, do you like the idea of the PR?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 14]
```diff
@@ -281,6 +281,54 @@
 	//You don't belong in this world, monster!
 	mytray.reagents.remove_reagent(type, volume)
 
+/datum/reagent/water/salt
+	name = "Saltwater"
+	description = "Water, but salty. Smells like... Central command shuttle docking?"
+	color = "#aaaaaa9d" // rgb: 170, 170, 170, 77 (alpha)
+	taste_description = "water"
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:288]: Should probably mention that it tastes salty.

**Comment by san7890** [comment]
> @san7890 Aside from the code nitpicks, do you like the idea of the PR?

i don't see why it's a bad idea given how common medical supplies are on a station already

File: code/datums/wounds/burns.dm (hunk 12 above)
Comment (carlarctg) [reply code/datums/wounds/burns.dm:43]: i dont see why not

File: code/__DEFINES/maths.dm [hunk 15]
```diff
@@ -240,6 +240,9 @@
 #define SPT_PROB(prob_per_second_percent, seconds_per_tick) (prob(100*SPT_PROB_RATE((prob_per_second_percent)/100, (seconds_per_tick))))
 // )
 
+// This value per these many units. Very unnecessary but helpful for readability (For example wanting 30 units of synthflesh to heal 50 damage - VALUE_PER(50, 30))
+#define VALUE_PER(value, per) value / per
```
Comment (MrMelbert) [inline code/__DEFINES/maths.dm:244]:
```suggestion
#define VALUE_PER(value, per) (value / per)
```
always wrap your macros

File: code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm [hunk 16]
```diff
@@ -663,10 +692,6 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 	default_container = /obj/item/reagent_containers/cup/glass/ice
 
-/datum/reagent/consumable/ice/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
-	affected_mob.adjust_bodytemperature(-5 * REM * TEMPERATURE_DAMAGE_COEFFICIENT * seconds_per_tick, affected_mob.get_body_temp_normal())
```
Comment (carlarctg) [inline code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:667]: wtf

**Comment by epochayur** [comment]
With all of Tea's functionality (it adjusts body temp AND heals tox AND heals wounds), it should really have an OD threshold at this point, seeing as people are already routinely going and drinking 3000u of tea roundstart BEFORE it gets a huge buff.

**Comment by Dorsisdwarf** [comment]
> With all of Tea's functionality (it adjusts body temp AND heals tox AND heals wounds), it should really have an OD threshold at this point, seeing as people are already routinely going and drinking 3000u of tea roundstart BEFORE it gets a huge buff.

OD makes you english

**Comment by RikuTheKiller** [comment]
Just realized a massive problem. This fucks up EVERY single chemical recipe buffer that most chemists have saved for the sake of simplifying a few chems. That's not worth it at all. Buffer recipes rely entirely on the pH of the ingredients and that will shift after this change, resulting in all chemists having to painstakingly redo a shit ton of work.

**Comment by carlarctg** [comment]
> Just realized a problem. This will fuck up buffers for purity stuff due to saltwater.

...How?

**Comment by RikuTheKiller** [comment]
> ...How?

Saltwater replaces chlorine + water, meaning all recipes that use chlorine + water will have the total pH of their ingredients altered. This means the other reagents you have to add to make the recipe result in a pure product will change and due to weird shit, it may change for other recipes as well. Basically, changing existing chemical recipes is risky.

In other words, chemists will have to redo a bunch of boring stuff they did ages ago, most of us keep around a list of buffers for specific recipes or use an online one, all of which will have to be partially redone after this PR for the sake of condensing a couple recipes.

**Comment by carlarctg** [comment]
> > ...How?
> 
> Saltwater replaces chlorine + water, meaning all recipes that use chlorine + water will have the total pH of their ingredients altered. This means the other reagents you have to add to make the recipe result in a pure product will change and due to weird shit, it may change for other recipes as well. Basically, changing existing chemical recipes is risky.
> 
> In other words, chemists will have to redo a bunch of boring stuff they did ages ago, most of us keep around a list of buffers for specific recipes or use an online one, all of which will have to be partially redone after this PR for the sake of condensing a couple recipes. Some online ones simply aren't maintained (or aren't maintained that well), so those will stay outdated for ages.
> 
> You could make saltwater require a bit of heat to react since salt needs heat to dissolve in water anyway. That's not hard to acquire through the use of a welder or an igniter and would leave all existing recipes intact. Saline doesn't need heating and is barely exothermic.

I'ma be real with you I just don't care enough about that, sorry. Fermichem is a cancer. You're free to suggest changes to my code or PR further ones but I can't bring myself to care about the fine tuning of purity.

**Comment by Loafin34** [comment]
> > 
> 
> 'Stabilize' in /tg/'s medical system is a bit meh because lethal vs non-lethal is based on how much damage numbers you have. If it's 100 or above, you ded. If not, you fine. /Tg/'s medic system is bizarre where things that'd be trivial irl can easily kill you (e.g., scratching at an itch! x10), irl scratching nonstop is harmful but it's not instantly deadly - it COULD technically kill you via long-term causes I guess, and OTOH things that'd be very dangerous IRL can be easily stabilized and treated like gunshot wounds, slashing open someone's arteries, having an entire armoured vehicle falling on your head, etc. In /tg/, you're more likely to die from 'oops I mislicked and touched a hot lightbulb when bandgaging myself at 99 damage' then the actual wounds which caused 99 damage. I could stabilize my wounds with epi, salt water, flour, dirt, inhaling cattle medicine and maybe the odd orange juice IV but I'd still die from a random itch, or from a mouse biting your jackbooted toe.
> 
> Also source on the tea curing toxin damage irl? If you had a very small dosage of poison you might be able to get away with it by drinking large quantities of tea because it'd be diluted, but this goes for water too. If you're actually dying from poison induced damage then I'm sorry but tea isn't going to help

we need a balance PR where rats cannot bite through jackboots

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 17]
```diff
@@ -380,6 +380,10 @@
 	. = ..()
 	metabolizer.remove_status_effect(/datum/status_effect/grouped/screwy_hud/fake_healthy, type)
 
+/datum/reagent/medicine/mine_salve/on_burn_wound_processing((datum/wound/burn/flesh/burn_wound))
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:383]:
```suggestion
/datum/reagent/medicine/mine_salve/on_burn_wound_processing(datum/wound/burn/flesh/burn_wound)
```

linters are failing

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by nikothedude** [comment]
FUCK

MY WOUNDS

**Pull request merged**